tests/README.md

```diff
@@ -0,0 +1,138 @@
+- **result**: The expected result of the function. Either `SUBSTRAIT_ERROR` or a literal value.
+- **literal**: In the format `<name>::<datatype>`
+- **description**: A string describing the test case
+
+    ```code
+    add(126::i8, 1::i8) = 127::i8  # addition of two numbers
+    ```
+
+### Spec
+
+```
+doc         := <version>
+               (<include>)+
+               ((<test_group>)?(<test_case>)+\n)+
+version     := ### SUBSTRAIT_SCALAR_TEST: <test_library_version>
+include     := ### SUBSTRAIT_INCLUDE: <uri>
+test_group  := # <description>
+test_case   := <function>(<arguments>) ([<options>])? = <result> (#<description>)?
+description := string
+function    := string
+arguments   := <argument>, <argument>, ... <argument>
+argument    := <literal>
+literal     := <name>::<datatype>
+result      := SUBSTRAIT_ERROR | <literal>
+options     := <optLiteral>, <optLiteral>, ... <optLiteral>
+optLiteral  := <option_name>:<option_value>
+```
+
+**TODO:** use ANTLR to describe the grammar and generate parser
+### Literals
+
+#### String
+- **string**, **fixedchar**, **varchar**: A sequence of characters enclosed in single quotes. Example: 'Hello, world!'
+
+#### Integer
+Integers are represented as sequences of digits. Negative numbers are preceded by a minus sign.
+- **i8**: 8-bit integer, range: -128 to 127
+- **i16**: 16-bit integer, range: -32,768 to 32,767
+- **i32**: 32-bit integer, range: -2,147,483,648 to 2,147,483,647
+- **i64**: 64-bit integer, range: -9,223,372,036,854,775,808 to 9,223,372,036,854,775,807
+
+#### Fixed Point decimals
+- **decimal**: Fixed-point decimal number. Maximum 38 digits total, with up to 37 digits after the decimal point.
+  Example: 123.456
+
+#### Floating Point numbers
+- **float**: General floating-point number, can be represented as:
+  * Standard decimal notation: 123.456
+  * Scientific notation: 1.23e4
+  * Special values: `nan` (Not a Number), `+inf` (Positive Infinity), `-inf` (Negative Infinity)
+- **float32**: Single-precision float, approximately 6 significant digits, range: ~1.2e-38 to ~3.4e38
+- **float64**: Double-precision float, approximately 15 significant digits, range: ~2.3e-308 to ~1.7e308
+
+#### Boolean
+- Valid values: TRUE, FALSE, NULL
+
+#### Date and Time
+All date and time literals use ISO 8601 format:
+
+- **date**: `YYYY-MM-DD`, example: `2021-01-01`
+- **time**: `HH:MM:SS[.fraction]`, example: `12:00:00.000`
+- **timestamp**: `YYYY-MM-DD HH:MM:SS[.fraction]`, example: `2021-01-01 12:00:00`
+- **timestamp_tz**: `YYYY-MM-DD HH:MM:SS[.fraction]±HH:MM`, example: `2021-01-01 12:00:00+05:30`
+- **interval year**: `INTERVAL 'P[n]Y[n]M'`, example: `INTERVAL 'P2Y3M'` (2 years, 3 months)
+- **interval days**: `INTERVAL 'P[n]DT[n]H[n]M[n]S'`, example: `INTERVAL 'P2DT3H2M9S'` (2 days, 3 hours, 2 minutes, 9 seconds)
```

Comment:
This a somewhat recent change, but also need to encode precision for interval_day now.

See https://substrait.io/types/type_classes/#compound-types